build: unbreak CPU feature selection

matklad opened the pull request:
This fixes the issue when, after Zig 0.13, TigerBeetle would sometimes fall back to software implementaiton of AEGIS.

What we need to do is to just enable a couple of CPU features. Previously, the logic worked by first performing normal target resolution in build.zig, and then patching the target to what we need.

However, after 0.13 the patching no longer works: the features is enabled in build.zig, but not in the actual binary. This is _probably_ due to discrepancy between Target.Query and ResolvedTarget, but I wasn't able to trace this to the root.

Instead, the entire logic is re-written to just pick one of the set of hard-coded targets.

As a bonus, `scripts -- release` now accepts optional `--target` and `--mode` flags, to compile only subset of release binaries.

cb22 commented:
Seems some stray rebase markers made it in!

matklad commented:
Oh, dear, I don't know what it means to build for a single platform with multiversioning :rofl: 

Backed off the release.zig changes for now!

cb22 commented on build.zig:
```diff
@@ -864,6 +870,7 @@ fn set_cpu_features(
     };
     target_requested.result.cpu.features.addFeatureSet(target_supported.cpu_features_add);
     target_requested.result.cpu.features.removeFeatureSet(target_supported.cpu_features_sub);
+    std.debug.print("{}\n", .{target_requested.result.cpu.features});
```
Are this and the above `print` meant to be here?

cb22 commented on src/scripts/devhub.zig:
```diff
@@ -34,6 +34,11 @@ pub fn main(shell: *Shell, gpa: std.mem.Allocator, cli_args: CliArgs) !void {
     try shell.project_root.deleteFile("tigerbeetle");
 
+    try shell.zig(
+        \\build scripts -- release --build --run-number=200 --sha={sha}
```
Can we change this to 189 - the current release.zig asserts that it's being called with `0.15.4` as a target.

matklad committed:
build: unbreak CPU feature selection

This fixes the issue when, after Zig 0.13, TigerBeetle would sometimes
fall back to software implementaiton of AEGIS.

What we need to do is to just enable a couple of CPU features.
Previously, the logic worked by first performing normal target
resolution in build.zig, and then patching the target to what we need.

However, after 0.13 the patching no longer works: the features is
enabled in build.zig, but not in the actual binary. This is _probably_
due to discrepancy between Target.Query and ResolvedTarget, but I wasn't
able to trace this to the root.

Instead, the entire logic is re-written to just pick one of the set of
hard-coded targets.
